Recruit-Training/Basic-Recruit-Training/person/Person.cpp

```diff
@@ -0,0 +1,50 @@
+
+#include <iostream>
+#include <string>
+#include "Person.h"
+
+Person::Person() : name_(""), age_(nullptr) {}
+
+Person::~Person() {
+    delete this->age_;
+}
+
+Person::Person(const std::string& name, int age) {
+    this->name_ = name;
+    this->age_ = new int(age);
+}
+
+int Person::getAge() const {
+    return *age_;
+}
+
+const std::string& Person::getName() const {
+    return name_;
+}
+
+void Person::setName(const std::string& newName) {
+    this->name_ = newName;
+}
+
+void Person::setAge(int newAge) {
+    delete this->age_;
```

Comment:
Instead of deleting age_ and recreating it, you should change the value of age_.